src/main/java/at/hannibal2/skyhanni/features/mining/eventtracker/MiningEventDisplay.kt

```diff
@@ -0,0 +1,94 @@
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStrings
+import at.hannibal2.skyhanni.utils.SimpleTimeMark.Companion.asTimeMark
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+object MiningEventDisplay {
+    private val config get() = SkyHanniMod.feature.mining.miningEvent
+
+    private var display = mutableListOf<String>()
+
+    private val dwarvenEvents = mutableListOf<RunningEvent>()
+    private val crystalEvents = mutableListOf<RunningEvent>()
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!event.repeatSeconds(1)) return
+        updateDisplay()
+    }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!shouldDisplay()) return
+        config.position.renderStrings(display, posLabel = "True Farming Fortune")
+    }
+
+    private fun updateDisplay() {
+        val shouldShowDwarven = when {
+            config.showType == MiningEventConfig.ShowType.BOTH -> true
+            config.showType == MiningEventConfig.ShowType.DWARVEN -> true
+            config.showType == MiningEventConfig.ShowType.CURRENT && IslandType.DWARVEN_MINES.isInIsland() -> true
+            else -> false
+        }
+        val shouldShowCrystal = when {
+            config.showType == MiningEventConfig.ShowType.BOTH -> true
+            config.showType == MiningEventConfig.ShowType.CRYSTAL -> true
+            config.showType == MiningEventConfig.ShowType.CURRENT && IslandType.CRYSTAL_HOLLOWS.isInIsland() -> true
+            else -> false
+        }
+
+        display.clear()
+        if (shouldShowDwarven) display.add("§aDwarven Mines: ${formUpcomingString(dwarvenEvents)}")
+        if (shouldShowCrystal) display.add("§aCrystal Hollows: ${formUpcomingString(crystalEvents)}")
+    }
+
+    private fun formUpcomingString(eventList: MutableList<RunningEvent>): String {
+        var output = ""
+        var foundCount = 0
+        for (event in eventList) {
+            val endsIn = event.endsAt.asTimeMark()
+            if (endsIn.isInPast()) continue
+
+            output = when (foundCount) {
+                0 -> event.event.toString()
+                else -> "$output §7-> ${event.event}"
+            }
+            if (event.isDouble) {
+                output = "$output §7-> ${event.event}"
+                foundCount ++
+            }
+            foundCount ++
+        }
+        if (output.isEmpty()) return "§8NONE"
+        return output
+    }
+
+    fun updateData(eventData: MiningEventData) {
+        dwarvenEvents.clear()
+        crystalEvents.clear()
+
+        for ((islandType, events) in eventData.runningEvents) {
+            when (islandType) {
+                IslandType.DWARVEN_MINES -> dwarvenEvents.addAll(events)
+                IslandType.CRYSTAL_HOLLOWS -> crystalEvents.addAll(events)
+                else -> continue
+            }
+        }
+    }
+
+    private fun shouldDisplay(): Boolean {
```

Comment:
this function does not need a single if. This is just an boolean expression, so please write it as one. Use linebraks to make it readable as it currently is.